rslint_core/src/groups/errors/use_isnan.rs

```diff
@@ -0,0 +1,221 @@
+                };
+                let op = expr.op().unwrap();
+
+                let always_text = if matches!(op, op!(!=) | op!(!==)) {
+                    "true"
+                } else {
+                    "false"
+                };
+
+                let mut err = ctx
+                    .err(
+                        self.name(),
+                        format!(
+                            "comparing `{}` to `NaN` using `{}` will always return {}",
+                            opposite.text(),
+                            expr.op_token().unwrap().text(),
+                            always_text
+                        ),
+                    )
+                    .primary(expr.range(), "")
+                    .note("note: `NaN` is not equal to anything including itself");
+
+                // telling the user to use isNaN for `<`, `>`, etc is a bit misleading so we won't do it if that is the case
+                if op == op!(==) || op == op!(===) {
+                    err = err.note(format!(
+                        "help: use `isNaN` instead: `{}`",
+                        color(&format!("isNaN({})", opposite))
+                    ))
+                } else if op == op!(!=) || op == op!(!==) {
+                    err = err.note(format!(
+                        "help: use `isNaN` instead: `{}`",
+                        color(&format!("!isNaN({})", opposite))
+                    ))
+                }
+
+                ctx.add_err(err);
+            }
+            SWITCH_STMT if self.enforce_for_switch_case => {
+                let stmt = node.to::<SwitchStmt>();
+                let expr = stmt.test()?.condition()?;
+                if expr.text() == "NaN" {
+                    let err = ctx
+                        .err(
+                            self.name(),
+                            "a switch statement with a test of `NaN` will never match",
+                        )
+                        .primary(expr.range(), "")
+                        .note("note: `NaN` is not equal to anything including itself");
+
+                    ctx.add_err(err);
+                }
+            }
+            CASE_CLAUSE if self.enforce_for_switch_case => {
+                let case = node.to::<CaseClause>();
+                let expr = case.test()?;
+                if expr.text() == "NaN" {
+                    let err = ctx
+                        .err(self.name(), "a case with a test of `NaN` will never match")
+                        .primary(expr.range(), "")
+                        .note("note: `NaN` is not equal to anything including itself");
+
+                    ctx.add_err(err);
+                }
+            }
+            CALL_EXPR if self.enforce_for_index_of => {
+                let expr = node.to::<CallExpr>();
+                let callee = expr.callee()?;
+                if is_indexof_static_prop(&callee)
+                    && expr.arguments()?.args().next()?.text() == "NaN"
```

Comment:
If you also forbid `Array.prototype.indexOf`, then the first argument is not the index but the array / string.

Reply:
There's also `String.prototype.indexOf`

Reply:
I will add specific handlers for those cases